### DISCUSSION TITLE
IDE Startup time

### DISCUSSION THREAD

**Discussion by wmaurer** [eclipse-theia/theia#14253] [Q&A] [answered]
Hi, I've started experimenting with Theia and find the idea very promising. 

My first reaction on starting the IDE was that the the startup time is rather slow compared to VSCode, and especially coming from Neovim (with a very sophisticated setup with many language servers, but lazy-loaded: [Astro Nvim](https://astronvim.com/))

A very basic Theia setup with core/electron/preferences seems to start quickly, but the full IDE setup starts very slowly.
 
My first thought was to to try out the browser version, but:
1) the frontend still seems to start rather slow
2) the frontend doesn't seem to be getting any info from the Typescript LSP

So my questions are:
1) Should separating the startup of the backend (e.g. on machine startup) and the frontend webapp be an approach to improve performance?
2) Is LSP support supposed to work with the browser IDE version?

Appreciate any help or ideas here 😄

**Comment by msujew** [comment]
Hey @wmaurer,

We have an epic for tracking startup performance improvements https://github.com/eclipse-theia/theia/issues/12924

> Should separating the startup of the backend (e.g. on machine startup) and the frontend webapp be an approach to improve performance?

I don't think this makes a lot of sense honestly. Or rather, this sounds pretty intrusive and I wouldn't want Theia to act that way if I were to install it on my machine.

> Is LSP support supposed to work with the browser IDE version?

It's supposed to work. Can you confirm that the TypeScript extension is installed in your IDE? Furthermore, do you get TypeScript support in the [online test environment](https://try.theia-cloud.io/)?

**Comment by JonasHelming** [comment]
@wmaurer Could you specify what "slow" means to you and also which OS and packaging you are using particularly for the Theia IDE?

**Comment by wmaurer** [comment]
I have a lot of projects going on, so I open project workspaces often.
Slow for me means having to wait watching a spinner for many seconds before I can start typing or navigating.

I'm running Manjaro Linux (Gnome) and have installed Theia via this package: https://aur.archlinux.org/packages/theia-bin

Here are some rough screen recordings showing some comparisions:
Nvim (fastest, even though it's loading additional plugins that VSCode/Theia isn't: e.g. eslint, prettier, etc)
[nvim.webm](https://github.com/user-attachments/assets/98175e9f-c46d-470e-94db-9cc90ed32603)

VSCode:
[vscode.webm](https://github.com/user-attachments/assets/2a0e5a79-dba8-4924-a3c2-4297577bf0f2)

Theia:
[theia.webm](https://github.com/user-attachments/assets/4ff484e6-7e44-4ba8-b658-4ea662c1370b)
(there's also a problem with the main window not automatically coming to the front after the spinner disappears, but that doesn't really affect this speed comparison)

My feelings from this comparison:
* Nvim is very fast (Time-To-Code would be faster if I set it up to remember the last opened file). With Nvim, I feel I'm productive - coding or searching after only 5 seconds.
* VSCode is *just* acceptable to me
* Theia feels sluggish, a bit like the old days with Visual Studio IDE or Eclipse

My wish with my experiments with Theia is the opening time of VSCode.

If I can achieve that, my end end goals would be (and I'm willing to put the work into to achieve this):
* Nvim editing experience in monaco (https://github.com/vscode-neovim/vscode-neovim)
* Nvim keyboard navigability, including modal editing in input fields (e.g. file rename or copilot chat)
* A more sophisticated UI than Nvim with VSCode/Theia UI features
* VSCode/Theia LSP and plugin stability (which is what I sometimes fight against with Nvim)

**Reply by JonasHelming** [reply, marked as answer]
As a first step, could you upfrade to 1.54 https://theia-ide.org/#theiaide). There should be a start up boost after 1.52.
For me, time to coding is ~5 seconds

**Reply by Mailaender** [reply]
I agree that it is noticeable slower to start than Visual Studio Code. Is it because of the splash screen?

**Reply by sdirix** [reply]
The splash screen does not slow down the tool. It's an optional screen (enabled in the Theia IDE) which shows as long as there would have been a loading spinner in the main window. As soon as the main window loading spinner is gone, we close the splash screen and show the main window.

**Reply by JonasHelming** [reply]
That is correct, but AFAIR, there was an issue before 1.53 with the Splash screen affecting loading time

**Comment by wmaurer** [comment]
Thanks 1.54 is faster than previous versions.
